### PR TITLE
[IA-2543] Explicitly set helm versions in dockerfile

### PR DESCRIPTION
**Pull request by jdcanas**
We specify these versions in the Dockerfile now as well as in reference.conf.

The default behavior of helm pull is to pull the latest, so there's no guarantee that versions specified in `.conf` will even be pulled.

We specify these versions in the `.conf` as well because it allows easier development and hotfixing. You can overwrite these values without rebuilding the docker image this way.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -19,7 +19,6 @@ EXPOSE 8080
 EXPOSE 5050
 
 ENV GIT_HASH $GIT_HASH
-# TODO: remove this
 ENV HELM_DEBUG 1
```
Comment (jdcanas) [inline Dockerfile:22]: I decided this is useful enough to keep, and only is noisy during failures, which is very helpful to always have on.

File: Dockerfile [hunk 2]
```diff
@@ -43,11 +42,12 @@ RUN helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx && \
 # pulling `terra-app-setup` locally and add cert files to the chart.
 # Leonardo will install the chart from local version.
 # We are also cacheing charts so they are not downloaded with every helm-install
+# If you are changing any versions here, update it in the reference.conf
 RUN pushd /leonardo && \
-    helm pull terra-app-setup-charts/terra-app-setup --untar && \
-    helm pull galaxy/galaxykubeman --untar && \
-    helm pull terra/terra-app --untar && \
-    helm pull ingress-nginx/ingress-nginx --untar && \
+    helm pull terra-app-setup-charts/terra-app-setup --version 0.0.2 --untar && \
```
Comment (Qi77Qi) [inline Dockerfile]: wonder if it's better to keep these versions as `ENV` on top of the file

**Review by Qi77Qi — APPROVED** [review]

**Review by rtitle — APPROVED** [review]
:shipit: if tests pass

File: Dockerfile [hunk 3]
```diff
@@ -43,11 +42,12 @@ RUN helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx && \
 # pulling `terra-app-setup` locally and add cert files to the chart.
 # Leonardo will install the chart from local version.
 # We are also cacheing charts so they are not downloaded with every helm-install
+# If you are changing any versions here, update it in the reference.conf
```
Comment (Qi77Qi) [inline Dockerfile]: how about prefix `WARNING:` cuz it's kinda buried in a blob of text

Comment (rtitle) [reply Dockerfile]: +1

File: Dockerfile (hunk 2 above)
Comment (jdcanas) [reply Dockerfile]: I don't feel strongly either way. Could make it easier to find, but I personally search for `helm pull` when looking for these versions

Comment (kyuksel) [reply Dockerfile]: +1 on keeping the versions as ENV on top of the file for better visibility

**Comment by jdcanas** [comment]
Fairly suspicious failure in stop/start test... https://fc-jenkins.dsp-techops.broadinstitute.org/job/leonardo-fiab-test-runner/34807/console

I want to run it a few more times here and run it by itself locally

jenkins retest

**Comment by rtitle** [comment]
Same `start app` test failure..

This PR is the first time Leo tests are using the network policy from https://github.com/DataBiosphere/terra-app/pull/10, right? I wonder if there some issue in the `start` flow (e.g. network policy is blocking something it shouldn't..)

If this needs further debugging it might be good to remove `0.0.2` from the helm repo before Monday's release. Alternatively: maybe update this PR to use `TERRA_APP_SETUP_VERSION=0.0.1` and see if it works?

**Comment by rtitle** [comment]
fiab-start failed, looks like `consent`. jenkins retest

**Pull request merged**